drivers/scale_host.go

```diff
@@ -0,0 +1,386 @@
+				if err != nil {
+					return code, err
+				}
+				count++
+			}
+		} else if deleteOption == "leastRecent" {
+			for count < amount {
+				index = (int64(len(hostScalingGroup)) - count) - 1
+				code, err := deleteHost(hostScalingGroup[index]["id"].(string), hostCreateURL, httpClient)
+				if err != nil {
+					return code, err
+				}
+				count++
+			}
+		}
+	}
+
+	return http.StatusOK, nil
+}
+
+func (s *ScaleHostDriver) ConvertToConfigAndSetOnWebhook(conf interface{}, webhook *model.Webhook) error {
+	if scaleConfig, ok := conf.(model.ScaleHost); ok {
+		webhook.ScaleHostConfig = scaleConfig
+		webhook.ScaleHostConfig.Type = webhook.Driver
+		return nil
+	} else if configMap, ok := conf.(map[string]interface{}); ok {
+		config := model.ScaleHost{}
+		err := mapstructure.Decode(configMap, &config)
+		if err != nil {
+			return err
+		}
+		webhook.ScaleHostConfig = config
+		webhook.ScaleHostConfig.Type = webhook.Driver
+		return nil
+	}
+	return fmt.Errorf("Can't convert config %v", conf)
+}
+
+func (s *ScaleHostDriver) GetDriverConfigResource() interface{} {
+	return model.ScaleHost{}
+}
+
+func (s *ScaleHostDriver) CustomizeSchema(schema *v1client.Schema) *v1client.Schema {
+	scaleOptions := []string{"up", "down"}
+	deleteOptions := []string{"mostRecent", "leastRecent"}
+	minValue := int64(1)
+
+	action := schema.ResourceFields["action"]
+	action.Type = "enum"
+	action.Options = scaleOptions
+	schema.ResourceFields["action"] = action
+
+	min := schema.ResourceFields["min"]
+	min.Default = 1
+	min.Min = &minValue
+	schema.ResourceFields["min"] = min
+
+	max := schema.ResourceFields["max"]
+	max.Default = 100
+	max.Min = &minValue
+	schema.ResourceFields["max"] = max
+
+	deleteOption := schema.ResourceFields["deleteOption"]
+	deleteOption.Type = "enum"
+	deleteOption.Options = deleteOptions
+	schema.ResourceFields["deleteOption"] = deleteOption
+
+	return schema
+}
+
+func getHosts(hostURL string) (map[string]interface{}, error) {
+	hostsResp := make(map[string]interface{})
+	resp, err := http.Get(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(respBytes, &hostsResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return hostsResp, nil
+}
+
+func createHost(host map[string]interface{}, hostCreateURL string, httpClient *http.Client) (int, error) {
+	hostJSON, err := json.Marshal(host)
+	if err != nil {
+		return http.StatusInternalServerError, fmt.Errorf("Error in JSON marshal of host : %v", err)
+	}
+
+	request, err := http.NewRequest("POST", hostCreateURL, bytes.NewBuffer(hostJSON))
+	if err != nil {
+		return http.StatusInternalServerError, fmt.Errorf("Error creating request : %v", err)
+	}
+
+	request.Header.Set("Content-Type", "application/json")
+	_, err = httpClient.Do(request)
+	if err != nil {
+		return http.StatusInternalServerError, fmt.Errorf("Error creating host : %v", err)
+	}
+
+	return http.StatusOK, nil
+}
+
```

Comment:
For deleteHost, for using API client, I would first have to get host ById. Since I'm already getting all hosts I thought I'll use http calls instead of getting hosts again and then deleting them.